### PR TITLE
feat(tasks): add a check for user existence

### PR DESCRIPTION
**Pull request by caspiano**
Task to check for user presence given domain and email.
Useful for setup automation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -83,6 +83,10 @@ By default, the backup will take place at midnight every day.
 
 - `help`: List all defined tasks
 
+- `check:user`: Check for existence of a user
+    * `domain`: Required
+    * `email`: Required
```
Comment (kimburgess) [inline README.md:88]:
```suggestion
    * `domain`: The PlaceOS domain the user is associated with (e.g. `example.com`). Required.
    * `email`: Email of the user (e.g. `alice@example.com`). Required.
```
(just to remove ambiguity over domain vs authority ID)

File: src/tasks/entities.cr [hunk 2]
```diff
@@ -8,6 +8,13 @@ module PlaceOS::Tasks::Entities
   extend self
   Log = ::Log.for(self)
 
+  # Check if a user exists under a domain
+  def user_exists?(email : String, domain : String) : Bool
+    !!Model::Authority.find_by_domain(domain).try do |authority|
+      Model::User.find_by_email(authority.id.as(String), email)
+    end
```
[the remark this answers is not in the document]
Comment (caspiano) [reply src/tasks/entities.cr]: Will reduce the start-up time, so I'll make the fix

[the remark this answers is not in the document]
Comment (caspiano) [reply src/tasks/entities.cr]: Will need to fetch the `Authority` and check its domain. I think this query is fine enough for this script

File: src/sam.cr [hunk 3]
```diff
@@ -68,6 +68,15 @@ namespace "drop" do
   end
 end
 
+namespace "check" do
+  desc "Check if a user exists on a domain"
+  task "user" do |_, args|
+    email = (args["email"]? || abort "`email` not set").to_s
+    domain = (args["domain"]? || abort "`domain` not set").to_s
+    exit(PlaceOS::Tasks.user_exists?(email, domain) ? 0 : 1)
```
[the remark this answers is not in the document]
Comment (caspiano) [reply src/sam.cr:76]: it's essentially a boolean check, I think I'd prefer a pretty clear `0` or `1` exit code in this case

File: src/tasks/entities.cr (hunk 2 above)
[the remark this answers is not in the document]
Comment (kimburgess) [reply src/tasks/entities.cr]: Ah yep - `User.find_by_email` is scoped by authority.

File: src/sam.cr (hunk 3 above)
[the remark this answers is not in the document]
Comment (kimburgess) [reply src/sam.cr:76]: FWIW this still provides that. Anything non-zero == error.

**Review by kimburgess — APPROVED** [review]

**Pull request merged**